File: android/src/com/tumblr/svetylk0/sheepsgohome/android/AndroidLauncher.java
package com.tumblr.svetylk0.sheepsgohome.android;

import android.content.ActivityNotFoundException;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.provider.Settings;
import android.telephony.TelephonyManager;

import com.badlogic.gdx.backends.android.AndroidApplication;
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration;
import com.sheepsgohome.SheepsGoHomeMain;
import com.sheepsgohome.interfaces.FunctionsInterface;

import java.util.Locale;

public class AndroidLauncher extends AndroidApplication implements FunctionsInterface{
	@Override
	protected void onCreate (Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
        //getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
		AndroidApplicationConfiguration config = new AndroidApplicationConfiguration();
		initialize(new SheepsGoHomeMain(this), config);
	}

    @Override
    public void launchDonateAction() {

    }

    @Override
    public void launchRateAppAction() {
        String packageName = "com.tumblr.svetylk0.sheepsgohome.android";
        Uri uri = Uri.parse("market://details?id=" + packageName);
        Intent goToMarket = new Intent(Intent.ACTION_VIEW, uri);
        try {
            startActivity(goToMarket);
        } catch (ActivityNotFoundException e) {
            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("http://play.google.com/store/apps/details?id=" + packageName)));
        }
    }

    @Override
    public String getDeviceId() {
        final TelephonyManager tm = (TelephonyManager) getBaseContext().getSystemService(Context.TELEPHONY_SERVICE);
        String id = tm.getDeviceId();

        if (id == null || id.equals("")) {
            id = Settings.Secure.getString(getContext().getContentResolver(),
                    Settings.Secure.ANDROID_ID);
        }

        return id;
    }

    @Override
    public String getCountryCode() {
        TelephonyManager tm = (TelephonyManager)getSystemService(Context.TELEPHONY_SERVICE);
        String country = tm.getSimCountryIso();

        if (country == null || country.equals("")) {
            country = Locale.getDefault().getCountry();
        }

        return country;
    }
}

File: android/src/com/tumblr/svetylk0/sheepsgohome/android/AndroidLauncher.kt
package com.tumblr.svetylk0.sheepsgohome.android

import android.content.Intent
import android.os.Bundle
import com.badlogic.gdx.backends.android.AndroidApplication
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration
import com.sheepsgohome.SheepsGoHomeMain
import com.sheepsgohome.shared.GameData
import com.tumblr.svetylk0.sheepsgohome.android.ActivityResult.RESOLVE_CONNECTION_ISSUE
import com.tumblr.svetylk0.sheepsgohome.android.libgdxbridge.AndroidBridge
import com.tumblr.svetylk0.sheepsgohome.android.libgdxbridge.GoogleClient
import com.tumblr.svetylk0.sheepsgohome.android.libgdxbridge.GoogleLeaderboardBridge
import com.tumblr.svetylk0.sheepsgohome.android.libgdxbridge.GoogleMultiplayerBridge

class AndroidLauncher : AndroidApplication() {

    lateinit var leaderboardBridge: GoogleLeaderboardBridge

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val config = AndroidApplicationConfiguration()

        GoogleClient.activity = this

        leaderboardBridge = GoogleLeaderboardBridge()

        GameData.android = AndroidBridge(this)
        GameData.leaderboard = leaderboardBridge
        GameData.multiplayer = GoogleMultiplayerBridge(this)

        initialize(SheepsGoHomeMain(), config)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        when(requestCode) {
            RESOLVE_CONNECTION_ISSUE -> GoogleClient.resolveConnectionIssue(resultCode)
            ActivityResult.MULTIPLAYER_WAITING_ROOM_INTENT -> GameData.multiplayer?.waitingRoomResult(resultCode)
        }
    }


}
